### ISSUE TITLE
FaceAnalysis.__init__() got an unexpected keyword argument 'providers'

### ISSUE THREAD

**Issue by Bigmeng1227** [cubiq/ComfyUI_IPAdapter_plus#271]
Error occurred when executing InsightFaceLoader:
FaceAnalysis.__init__() got an unexpected keyword argument 'providers'
File "D:\ComfyUI-aki-v1.1\execution.py", line 152, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "D:\ComfyUI-aki-v1.1\execution.py", line 82, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "D:\ComfyUI-aki-v1.1\execution.py", line 75, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "D:\ComfyUI-aki-v1.1\custom_nodes\ComfyUI_IPAdapter_plus\IPAdapterPlus.py", line 633, in load_insight_face
model = FaceAnalysis(name="buffalo_l", root=INSIGHTFACE_DIR, providers=[provider + 'ExecutionProvider',])
![微信截图_20240222135103](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/159002871/da27c547-3773-4f1b-b432-edb0f3cdf0c0)
![微信截图_20240222135133](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/159002871/2f266756-6ab6-4ac7-96b7-cd6fcb3af3fe)
是不是只有我一个人遇到了这个问题？？？

**Comment by cubiq** [comment]
you probably have an old workflow and a newer version of the extension, or vice-versa.

update the extension. refresh the browser. delete the IPAdapter nodes and recreate them

**Comment by Bigmeng1227** [comment]
> you probably have an old workflow and a newer version of the extension, or vice-versa.
> 
> update the extension. refresh the browser. delete the IPAdapter nodes and recreate them

OK thanks，I'll try your method right now！

**Comment by Bigmeng1227** [comment]
> you probably have an old workflow and a newer version of the extension, or vice-versa.
> 
> update the extension. refresh the browser. delete the IPAdapter nodes and recreate them

I‘m sorry to tell you, it dosen't work~

**Comment by Bigmeng1227** [comment]
> you probably have an old workflow and a newer version of the extension, or vice-versa.
> 
> update the extension. refresh the browser. delete the IPAdapter nodes and recreate them

![微信截图_20240222154837](https://github.com/cubiq/ComfyUI_IPAdapter_plus/assets/159002871/622999cf-cffd-4c91-b14c-bcd99cf1a898)
Is the error related to this file? Is it stored in the correct location?

**Comment by cubiq** [comment]
can you try in an incognito windows?

**Comment by lucakempkes** [comment]
pip install insightface==0.7.3 worked for me

**Comment by Bigmeng1227** [comment]
> pip install insightface==0.7.3 worked for me

C:\Users\asus>pip show insightface
Name: insightface
Version: 0.7.3
Summary: InsightFace Python Library
Home-page: https://github.com/deepinsight/insightface
Author: InsightFace Contributors
Author-email: contact@insightface.ai
License: MIT
Location: c:\users\asus\appdata\local\programs\python\python310\lib\site-packages
Requires: albumentations, cython, easydict, matplotlib, numpy, onnx, Pillow, prettytable, requests, scikit-image, scikit-learn, scipy, tqdm
Required-by:
Mine is also this version

**Comment by cubiq** [comment]
closing for inactivity

**Comment by cse84** [comment]
Just for posterity: You will get the error in OP's post if you install an old version of Insightface (I think I tried 0.2.0). ComfyUI_IPAdapter_plus doesn't work with that version (I don't think even the oldest commit works with that version). I only gave this old version of Insightface a try because trying to install the newest version gave me such troubles (and because ComfyUI_IPAdapter_plus doesn't specify which version exactly is needed).
